### PR TITLE
feat: Add spec.manager

### PR DESCRIPTION
**Pull request by ruanxin**
**Description**

**Related issue(s)**
#1840

### CODE REVIEW

File: api/v1beta2/moduletemplate_types.go [hunk 1]
```diff
@@ -138,11 +138,34 @@ type ModuleTemplateSpec struct {
 
 	// Info contains metadata about the module.
 	// +optional
-	Info ModuleInfo `json:"info,omitempty"`
+	Info *ModuleInfo `json:"info,omitempty"`
```
Comment (c-pius) [inline api/v1beta2/moduletemplate_types.go:141]: Why is this change necessary?

File: api/v1beta2/moduletemplate_types.go [hunk 2]
```diff
@@ -138,11 +138,34 @@ type ModuleTemplateSpec struct {
...
 
 	// AssociatedResources is a list of module related resources that usually must be cleaned when uninstalling a module. Informational purpose only.
 	// +optional
 	AssociatedResources []apimetav1.GroupVersionKind `json:"associatedResources,omitempty"`
+
+	// Manager contains information identifying a module's resource that indicates the readiness of the module. Typically, this should be the manager Deployment of the module.
```
Comment (c-pius) [inline api/v1beta2/moduletemplate_types.go:147]:
```suggestion
	// Manager contains information for identifying a module's resource that can be used as indicator for the installation readiness of the module. Typically, this is the manager Deployment of the module. In exceptional cases, it may also be another resource. 
```

File: api/v1beta2/moduletemplate_types.go [hunk 3]
```diff
@@ -138,11 +138,34 @@ type ModuleTemplateSpec struct {
...
+// Manager defines the structure for the manager field in ModuleTemplateSpec.
+type Manager struct {
+	// Group is the API group of the manager.
+	Group string `json:"group"`
+
+	// Version is the API version of the manager.
+	Version string `json:"version"`
+
+	// Kind is the kind of the manager.
+	Kind string `json:"kind"`
```
Comment (c-pius) [inline api/v1beta2/moduletemplate_types.go:161]: Maybe it would be possible to re-use the K8s `GroupVersionKind` type (see also AssociatedResources field). We would need to add a marker that inlines the field: `json:",inline"`.

But I am also okay using them as is. For Namespace and Name I haven't found a proper existing type that can be re-used.

File: docs/contributor/resources/03-moduletemplate.md [hunk 4]
```diff
@@ -75,6 +75,38 @@ spec:
 - documentation: The link to the documentation of the module.
 - icons: A list of icons of the module, each with a name and link.
 
+### **.spec.manager**
+
+The `manager` field provides details about a module's resource that indicates its readiness. Typically, this is the
+`Deployment` managing the module. If the resource is not namespaced, such as the module CR's `CRD`, the namespace can be
+optional.
```
Comment (c-pius) [inline docs/contributor/resources/03-moduletemplate.md:82]:
```suggestion
The `manager` field provides information for identifying a module's resource that can be used as indicator for the installation readiness of the module. Typically, this is the manager Deployment of the module. In exceptional cases, it may also be another resource. If the resource is not namespaced, such as the module CR's `CRD`, the namespace can be optional.
```

File: docs/contributor/resources/03-moduletemplate.md [hunk 5]
```diff
@@ -75,6 +75,38 @@ spec:
...
+For example, if the module has a manager `Deployment` in the `kyma-system` namespace:
+
+```yaml
+spec:
+  manager:
+    group: apps
+    version: v1
+    kind: Deployment
+    namespace: kyma-system
+    name: [module manager]
```
Comment (c-pius) [inline docs/contributor/resources/03-moduletemplate.md:93]:
```suggestion
    name: [module manager name]
```

File: docs/contributor/resources/03-moduletemplate.md [hunk 6]
```diff
@@ -75,6 +75,38 @@ spec:
...
+
+For a module without a manager, such as [cloud manager](https://github.com/kyma-project/cloud-manager):
+
+```yaml
+spec:
+  manager:
+    group: apiextensions.k8s.io
+    version: v1
+    kind: CustomResourceDefinition
+    name: [module CRD]
```
Comment (c-pius) [inline docs/contributor/resources/03-moduletemplate.md:104]:
```suggestion
    name: [module CRD name]
```

File: docs/contributor/resources/03-moduletemplate.md [hunk 7]
```diff
@@ -75,6 +75,38 @@ spec:
...
+```
+
+This field helps identify the module's manager, which is useful for determining whether a module is successfully
+installed, especially in manual installation scenarios.
```
Comment (c-pius) [inline docs/contributor/resources/03-moduletemplate.md:108]: I think this should be obvious from the above description. If we consider this to be helpful, I would at least move it up to the rest of the description.

**Review by c-pius — COMMENTED** [review]
I was thinking a bit about the name as well. I think staying with `manager` is okay as we also describe that this is our primary expected case. Also, I still have my doubts if Cloud Manager will even have a use case for manual installation. So I would say it is okay. If we really find that the name is bad, I think we could also rename it in a future API version or even introduce a conversion webhook replacing it with another field name.

File: api/v1beta2/moduletemplate_types.go (hunk 1 above)
Comment (ruanxin) [reply api/v1beta2/moduletemplate_types.go:141]: use pointer will make the actual code create this `ModuleInfo` a bit easier, with `nil` providing null field, and not necessarily creating an empty `ModuleInfo` struct.

File: docs/contributor/resources/03-moduletemplate.md (hunk 7 above)
Comment (ruanxin) [reply docs/contributor/resources/03-moduletemplate.md:106]: agree, removed

File: api/v1beta2/moduletemplate_types.go (hunk 3 above)
Comment (ruanxin) [reply api/v1beta2/moduletemplate_types.go]: good point, refactored

File: docs/contributor/resources/03-moduletemplate.md [hunk 8]
```diff
@@ -75,6 +75,32 @@ spec:
...
+For a module without a manager, such as [cloud manager](https://github.com/kyma-project/cloud-manager):
+
+```yaml
+spec:
+  manager:
+    group: apiextensions.k8s.io
+    version: v1
+    kind: CustomResourceDefinition
+    name: [module CRD name]
+```
```
Comment (c-pius) [inline docs/contributor/resources/03-moduletemplate.md:103]:
```suggestion
```

Thinking about it, maybe we should remove this example entirely. Just to not give the impression that this is something that should be done.

**Review by c-pius — DISMISSED** [review]
Just one last thought but also LGTM as is.

File: docs/contributor/resources/03-moduletemplate.md (hunk 8 above)
Comment (ruanxin) [reply docs/contributor/resources/03-moduletemplate.md:103]: if we need to have this use case, I think we should mention it, for documentation purpose, it also brings value,  implicitely as hidden rule which applied for certain modules could lead to more confusion.

However, as for whether it make sense of making CRD as the manager, we can open a dicussion on this. It might be better to leave this field optional, assuming that for modules without a manager, they are considered ready by default. Until a decision is made, I believe we should keep this use case as a valid example

File: docs/contributor/resources/03-moduletemplate.md [hunk 9]
```diff
@@ -75,6 +75,32 @@ spec:
 - documentation: The link to the documentation of the module.
 - icons: A list of icons of the module, each with a name and link.
 
+### **.spec.manager**
+
+The `manager` field provides information for identifying a module's resource that can be used as indicator for the installation readiness of the module. Typically, this is the manager Deployment of the module. In exceptional cases, it may also be another resource. If the resource is not namespaced, such as the module CR's `CRD`, the namespace can be optional.
```
Comment (mmitoraj) [inline docs/contributor/resources/03-moduletemplate.md:80]:
```suggestion
The `manager` field provides information for identifying a module's resource that can indicate the module's installation readiness. Typically, the manager is the module's `Deployment` resource. In exceptional cases, it may also be another resource. The **namespace** parameter is optional if the resource is not namespaced. For example,  if the resource is the module CR's `CustomResourceDefinition`.
```

File: docs/contributor/resources/03-moduletemplate.md [hunk 10]
```diff
@@ -75,6 +75,32 @@ spec:
...
+
+For example, if the module has a manager `Deployment` in the `kyma-system` namespace:
```
Comment (mmitoraj) [inline docs/contributor/resources/03-moduletemplate.md:82]:
```suggestion
In this example, the module's manager is the `Deployment` resource in the `kyma-system` namespace.
```

File: docs/contributor/resources/03-moduletemplate.md [hunk 11]
```diff
@@ -75,6 +75,32 @@ spec:
...
+spec:
+  manager:
+    group: apps
+    version: v1
+    kind: Deployment
+    namespace: kyma-system
+    name: [module manager name]
+```
+
+For a module without a manager, such as [cloud manager](https://github.com/kyma-project/cloud-manager):
```
Comment (mmitoraj) [inline docs/contributor/resources/03-moduletemplate.md:94]:
```suggestion
In this example, the module doesn't have a manager and in the **manager** field you specifie the module's `CustomResourceDefinition`. For example, of the [Cloud Manager](https://github.com/kyma-project/cloud-manager) module:
```
I'm just not sure if it means that there is no manager or if the manager is the CRD?

File: docs/contributor/resources/03-moduletemplate.md [hunk 12]
```diff
@@ -75,6 +75,32 @@ spec:
...
+spec:
+  manager:
+    group: apps
+    version: v1
+    kind: Deployment
+    namespace: kyma-system
+    name: [module manager name]
+```
+
+In this example, the module doesn't have a manager and in the **manager** field you specifie the module's `CustomResourceDefinition`. For example, of the [Cloud Manager](https://github.com/kyma-project/cloud-manager) module:
```
Comment (mmitoraj) [inline docs/contributor/resources/03-moduletemplate.md:94]:
```suggestion
In this example, the module's manager is the module's `CustomResourceDefinition` that does not require the **namespace** parameter.
```

**Review by c-pius — APPROVED** [review]

**Review by mmitoraj — APPROVED** [review]
